include/ddc/create_mirror.hpp

```diff
@@ -13,118 +13,117 @@
+/// Equivalent to `create_mirror(Kokkos::HostSpace(), src)`.
+/// @param[in] src A layout right ChunkSpan.
+/// @return a `Chunk` with the same support and layout as `src` allocated on the `Kokkos::HostSpace` memory space.
 template <class ElementType, class Support, class Layout, class MemorySpace>
 auto create_mirror(ChunkSpan<ElementType, Support, Layout, MemorySpace> const& src)
 {
-    static_assert(
-            std::is_same_v<Layout, std::experimental::layout_right>,
-            "Only layout right is supported");
-    return Chunk(src.domain(), HostAllocator<std::remove_const_t<ElementType>>());
+    return create_mirror(Kokkos::HostSpace(), src);
 }
 
-/// Returns a new `Chunk` with the same layout as `src` allocated on the memory space `Space::memory_space` and operates a deep copy between the two.
+/// @param[in] space A Kokkos memory space or execution space.
+/// @param[in] src A layout right ChunkSpan.
+/// @return a `Chunk` with the same support and layout as `src` allocated on the `Space::memory_space` memory space and operates a deep copy between the two.
 template <class Space, class ElementType, class Support, class Layout, class MemorySpace>
 auto create_mirror_and_copy(
         Space const& space,
         ChunkSpan<ElementType, Support, Layout, MemorySpace> const& src)
 {
-    static_assert(
-            std::is_same_v<Layout, std::experimental::layout_right>,
-            "Only layout right is supported");
     Chunk chunk = create_mirror(space, src);
-    parallel_deepcopy(space, chunk, src);
+    parallel_deepcopy(chunk, src);
     return chunk;
 }
 
-/// Returns a new host `Chunk` with the same layout as `src` and operates a deep copy between the two.
+/// Equivalent to `create_mirror_and_copy(Kokkos::HostSpace(), src)`.
+/// @param[in] src A layout right ChunkSpan.
+/// @return a `Chunk` with the same support and layout as `src` allocated on the `Kokkos::HostSpace` memory space and operates a deep copy between the two.
 template <class ElementType, class Support, class Layout, class MemorySpace>
 auto create_mirror_and_copy(ChunkSpan<ElementType, Support, Layout, MemorySpace> const& src)
 {
-    static_assert(
-            std::is_same_v<Layout, std::experimental::layout_right>,
-            "Only layout right is supported");
-    Chunk chunk = create_mirror(src);
-    parallel_deepcopy(chunk, src);
-    return chunk;
+    return create_mirror_and_copy(Kokkos::HostSpace(), src);
 }
 
-/// If `src` is accessible from `space` then returns a copy of `src`,
-/// otherwise returns a new `Chunk` with the same layout as `src` allocated on the memory space `Space::memory_space`.
+/// @param[in] space A Kokkos memory space or execution space.
+/// @param[in] src A non-const, layout right ChunkSpan.
+/// @return If `MemorySpace` is accessible from `Space` then returns a copy of `src`, otherwise returns a `Chunk` with the same support and layout as `src` allocated on the `Space::memory_space` memory space.
```

Comment:
Not sure if you care about line length in DDC?
```suggestion
/// @return If `MemorySpace` is accessible from `Space` then returns a copy of `src`, otherwise returns a `Chunk`
///         with the same support and layout as `src` allocated on the `Space::memory_space` memory space.
```

Reply:
It makes me realize we don't have enabled `ReflowComments` in clang-format. I guess we should do it.

Reply:
Probably in a separate PR though I guess?

Reply:
Yep